### PR TITLE
Branch integrate job company easy way

### LINKED ISSUE #90: Integrate Job and Company together

### PR DESCRIPTION
**Pull request by Green-Tea-123**
closes #90 
This integrates the job and company classes by having the name of the company be an attribute of a job. The check for company existence is implemented in the model's addJob function. This handles all of the existence checking when loading from save as well. I determined that company need not have a collection of jobnames as the model could know if a job was from a company if it just compared the company's name to the companyname in job.

Potential bug to fix:
When I throw the CompanyNotFoundException in model, the exception is caught and displayed in the addressbook. However, the terminal still displays an exception called TargetInvocationException that I'm not sure of the origin.

### CODE REVIEW

**Review by wassilililily — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/model/AddressBook.java [hunk 1]
```diff
@@ -113,8 +116,17 @@ public void addPerson(Person p) {
...
 
-    /** Adds a job to the address book. */
+    /**
+     * Adds a job to the address book.
+     * The existence of the company referenced by the job creation is checked here.
+     */
     public void addJob(Job j) {
+        boolean companyExists = StreamSupport
+                .stream(companies.spliterator(), false)
+                .anyMatch(x -> j.getCompany().matchesCompanyName(x.getName()));
```
Comment (Quasant) [inline src/main/java/seedu/address/model/AddressBook.java:126]: Efficiency related -> j.getCompany() can be computed once and saved to a variable.

File: src/main/java/seedu/address/model/job/Job.java [hunk 2]
```diff
@@ -25,6 +27,8 @@ public class Job {
      */
     public Job(Name name, JobCompany company, JobSalary salary,
             JobDescription description, Set<Tag> requirements) {
+        requireAllNonNull(name, company, salary, description);
+        requireAllNonNull(requirements);
```
Comment (Quasant) [inline src/main/java/seedu/address/model/job/Job.java:31]: Is it possible to put requirements inside the first requireAllNonNull()?

**Review by Quasant — APPROVED** [review]
Yea most things LGTM.

File: src/main/java/seedu/address/model/AddressBook.java (hunk 1 above)
Comment (Green-Tea-123) [reply src/main/java/seedu/address/model/AddressBook.java:126]: Ok I can resolve that

File: src/main/java/seedu/address/model/job/Job.java (hunk 2 above)
Comment (Quasant) [reply src/main/java/seedu/address/model/job/Job.java:31]: Ah I see, there might be some big implications, although I am not sure how a `null` value would enter the collection.

File: src/main/java/seedu/address/model/AddressBook.java (hunk 1 above)
Comment (Quasant) [reply src/main/java/seedu/address/model/AddressBook.java:126]: Hahhaa its optional, just a small nit.

**Pull request merged**